[bug] cmake find_program regression in conan 1.54+

### Environment details

* Operating System+version: Windows 10 21H2 19044.2251
* Compiler+version: MSVC 19.29.30146.0
* Conan version: 1.54, 1.57
* Python version: 3.10.4


### Steps to reproduce

1. Download zip file: [conan_find_program.zip](https://github.com/conan-io/conan/files/10480121/conan_find_program.zip)
2. Unzip to directory
3. run `conan create .`
4. Observe that with conan 1.53 there is no error, but with conan 1.54 hello_exe can not be found.


We had to upgrade to 1.54 due to some conan-center recipes requiring this version. When we tried , we encountered problems using `find_program` to find `protoc` from `protobuf` as well as `capnp` from `capnproto`. Suddenly they could not be found anymore, even though they clearly are in the package folder.

I made this minimal repro based on the `conan new` cmake templates, and we can see that in 1.53 find_program finds hello_exe no problem, but as soon as you upgrade to 1.54, it is no longer found.

### Logs

**NON WORKING (1.54)**
```
D:\conan_find_program>conan create .
WARN: Migration: Updating settings.yml
Migration: Settings already up to date
Exporting package recipe
hello/0.1 exports_sources: Copied 1 '.txt' file: CMakeLists.txt
hello/0.1 exports_sources: Copied 2 '.cpp' files: hello.cpp, main.cpp
hello/0.1 exports_sources: Copied 1 '.h' file: hello.h
hello/0.1: The stored package has not changed
hello/0.1: Using the exported files summary hash as the recipe revision: 0dbe589a6e52ddaa9ab052a116cde54b
hello/0.1: Exported revision: 0dbe589a6e52ddaa9ab052a116cde54b
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=Visual Studio
compiler.runtime=MD
compiler.version=16
os=Windows
os_build=Windows
[options]
[build_requires]
[env]

hello/0.1: Forced build from source
hello/0.1 (test package): Installing package
Requirements
    hello/0.1 from local cache - Cache
Packages
    hello/0.1:3fb49604f9c2f729b85ba3115852006824e72cab - Build

Installing (downloading, building) binaries...
hello/0.1: Copying sources to build folder
hello/0.1: Building your package in C:\Users\my_username\.conan\data\hello\0.1\_\_\build\3fb49604f9c2f729b85ba3115852006824e72cab
hello/0.1: Generator txt created conanbuildinfo.txt
hello/0.1: Calling generate()
hello/0.1: WARN: Using the new toolchains and generators without specifying a build profile (e.g: -pr:b=default) is discouraged and might cause failures and unexpected behavior
hello/0.1: Preset 'default' added to CMakePresets.json. Invoke it manually using 'cmake --preset default'
hello/0.1: If your CMake version is not compatible with CMakePresets (<3.19) call cmake like: 'cmake <path> -G "Visual Studio 16 2019" -DCMAKE_TOOLCHAIN_FILE=C:\Users\my_username\.conan\data\hello\0.1\_\_\build\3fb49604f9c2f729b85ba3115852006824e72cab\build\generators\conan_toolchain.cmake -DCMAKE_POLICY_DEFAULT_CMP0091=NEW'
hello/0.1: Aggregating env generators
hello/0.1: Calling build()
hello/0.1: CMake command: cmake -G "Visual Studio 16 2019" -DCMAKE_TOOLCHAIN_FILE="C:/Users/my_username/.conan/data/hello/0.1/_/_/build/3fb49604f9c2f729b85ba3115852006824e72cab/build/generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="C:/Users/my_username/.conan/data/hello/0.1/_/_/package/3fb49604f9c2f729b85ba3115852006824e72cab" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" "C:\Users\my_username\.conan\data\hello\0.1\_\_\build\3fb49604f9c2f729b85ba3115852006824e72cab\."
-- Using Conan toolchain: C:/Users/my_username/.conan/data/hello/0.1/_/_/build/3fb49604f9c2f729b85ba3115852006824e72cab/build/generators/conan_toolchain.cmake
-- Selecting Windows SDK version 10.0.19041.0 to target Windows 10.0.19044.
-- The CXX compiler identification is MSVC 19.29.30146.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Professional/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_POLICY_DEFAULT_CMP0091


-- Build files have been written to: C:/Users/my_username/.conan/data/hello/0.1/_/_/build/3fb49604f9c2f729b85ba3115852006824e72cab/build
hello/0.1: CMake command: cmake --build "C:\Users\my_username\.conan\data\hello\0.1\_\_\build\3fb49604f9c2f729b85ba3115852006824e72cab\build" --config Release
Microsoft (R) Build Engine version 16.11.2+f32259642 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

  Checking Build System
  Building Custom Rule C:/Users/my_username/.conan/data/hello/0.1/_/_/build/3fb49604f9c2f729b85ba3115852006824e72cab/CMakeLists.txt
  hello.cpp
  main.cpp
  hello_exe.vcxproj -> C:\Users\my_username\.conan\data\hello\0.1\_\_\build\3fb49604f9c2f729b85ba3115852006824e72cab\build\Release\hello_exe.
  exe
  Building Custom Rule C:/Users/my_username/.conan/data/hello/0.1/_/_/build/3fb49604f9c2f729b85ba3115852006824e72cab/CMakeLists.txt
  hello.cpp
  hello_lib.vcxproj -> C:\Users\my_username\.conan\data\hello\0.1\_\_\build\3fb49604f9c2f729b85ba3115852006824e72cab\build\Release\hello_lib.
  lib
  Building Custom Rule C:/Users/my_username/.conan/data/hello/0.1/_/_/build/3fb49604f9c2f729b85ba3115852006824e72cab/CMakeLists.txt
hello/0.1: Package '3fb49604f9c2f729b85ba3115852006824e72cab' built
hello/0.1: Build folder C:\Users\my_username\.conan\data\hello\0.1\_\_\build\3fb49604f9c2f729b85ba3115852006824e72cab\build
hello/0.1: Generated conaninfo.txt
hello/0.1: Generated conanbuildinfo.txt
hello/0.1: Generating the package
hello/0.1: Package folder C:\Users\my_username\.conan\data\hello\0.1\_\_\package\3fb49604f9c2f729b85ba3115852006824e72cab
hello/0.1: Calling package()
hello/0.1: CMake command: cmake --install "C:\Users\my_username\.conan\data\hello\0.1\_\_\build\3fb49604f9c2f729b85ba3115852006824e72cab\build" --config Release --prefix "C:/Users/my_username/.conan/data/hello/0.1/_/_/package/3fb49604f9c2f729b85ba3115852006824e72cab"
-- Installing: C:/Users/my_username/.conan/data/hello/0.1/_/_/package/3fb49604f9c2f729b85ba3115852006824e72cab/bin/hello_exe.exe
-- Installing: C:/Users/my_username/.conan/data/hello/0.1/_/_/package/3fb49604f9c2f729b85ba3115852006824e72cab/lib/hello_lib.lib
-- Installing: C:/Users/my_username/.conan/data/hello/0.1/_/_/package/3fb49604f9c2f729b85ba3115852006824e72cab/include/hello.h
hello/0.1 package(): Packaged 1 '.exe' file: hello_exe.exe
hello/0.1 package(): Packaged 1 '.h' file: hello.h
hello/0.1 package(): Packaged 1 '.lib' file: hello_lib.lib
hello/0.1: Package '3fb49604f9c2f729b85ba3115852006824e72cab' created
hello/0.1: Created package revision ec8be79497c56ac23819a642c5c0840f
hello/0.1 (test package): WARN: Using the new toolchains and generators without specifying a build profile (e.g: -pr:b=default) is discouraged and might cause failures and unexpected behavior
hello/0.1 (test package): Generator 'CMakeToolchain' calling 'generate()'
hello/0.1 (test package): Preset 'default' added to CMakePresets.json. Invoke it manually using 'cmake --preset default'
hello/0.1 (test package): If your CMake version is not compatible with CMakePresets (<3.19) call cmake like: 'cmake <path> -G "Visual Studio 16 2019" -DCMAKE_TOOLCHAIN_FILE=D:\conan_find_program\test_package\build\generators\conan_toolchain.cmake -DCMAKE_POLICY_DEFAULT_CMP0091=NEW'
hello/0.1 (test package): WARN: Using the new toolchains and generators without specifying a build profile (e.g: -pr:b=default) is discouraged and might cause failures and unexpected behavior
hello/0.1 (test package): Generator 'CMakeDeps' calling 'generate()'
hello/0.1 (test package): Generator txt created conanbuildinfo.txt
hello/0.1 (test package): Aggregating env generators
hello/0.1 (test package): Generated conaninfo.txt
hello/0.1 (test package): Generated graphinfo
Using lockfile: 'D:\conan_find_program\test_package\build\generators/conan.lock'
Using cached profile from lockfile
hello/0.1 (test package): Calling build()
hello/0.1 (test package): CMake command: cmake -G "Visual Studio 16 2019" -DCMAKE_TOOLCHAIN_FILE="D:/conan_find_program/test_package/build/generators/conan_toolchain.cmake" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" "D:\conan_find_program\test_package\."
-- Using Conan toolchain: D:/conan_find_program/test_package/build/generators/conan_toolchain.cmake
-- Selecting Windows SDK version 10.0.19041.0 to target Windows 10.0.19044.
-- The CXX compiler identification is MSVC 19.29.30146.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Professional/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Conan: Target declared 'hello::hello'
CMake Debug Log at CMakeLists.txt:7 (find_program):
  find_program called with the following settings:

    VAR: TEST
    NAMES: "hello_exe"
    Documentation: Path to a program.
    Framework
      Only Search Frameworks: 0
      Search Frameworks Last: 0
      Search Frameworks First: 0
    AppBundle
      Only Search AppBundle: 0
      Search AppBundle Last: 0
      Search AppBundle First: 0
    CMAKE_FIND_USE_CMAKE_PATH: 1
    CMAKE_FIND_USE_CMAKE_ENVIRONMENT_PATH: 1
    CMAKE_FIND_USE_SYSTEM_ENVIRONMENT_PATH: 1
    CMAKE_FIND_USE_CMAKE_SYSTEM_PATH: 1

  find_program considered the following locations:

    D:/conan_find_program/test_package/build/generators/bin/hello_exe.com
    D:/conan_find_program/test_package/build/generators/bin/hello_exe.exe
    D:/conan_find_program/test_package/build/generators/bin/hello_exe
    D:/conan_find_program/test_package/build/generators/sbin/hello_exe.com
    D:/conan_find_program/test_package/build/generators/sbin/hello_exe.exe
    D:/conan_find_program/test_package/build/generators/sbin/hello_exe
    D:/conan_find_program/test_package/build/generators/hello_exe.com
    D:/conan_find_program/test_package/build/generators/hello_exe.exe
    D:/conan_find_program/test_package/build/generators/hello_exe
    C:/.conan/6ab047/1/bin/hello_exe.com
    C:/.conan/6ab047/1/bin/hello_exe.exe
    C:/.conan/6ab047/1/bin/hello_exe
    C:/.conan/4867a5/1/bin/hello_exe.com
    C:/.conan/4867a5/1/bin/hello_exe.exe
    C:/.conan/4867a5/1/bin/hello_exe
    C:/Program Files/Python310/Scripts/hello_exe.com
    C:/Program Files/Python310/Scripts/hello_exe.exe
    C:/Program Files/Python310/Scripts/hello_exe
    C:/Program Files/Python310/hello_exe.com
    C:/Program Files/Python310/hello_exe.exe
    C:/Program Files/Python310/hello_exe
    C:/Program Files (x86)/Intel/oneAPI/mpi/latest/bin/hello_exe.com
    C:/Program Files (x86)/Intel/oneAPI/mpi/latest/bin/hello_exe.exe
    C:/Program Files (x86)/Intel/oneAPI/mpi/latest/bin/hello_exe
    C:/Program Files (x86)/Intel/oneAPI/mpi/latest/bin/release/hello_exe.com
    C:/Program Files (x86)/Intel/oneAPI/mpi/latest/bin/release/hello_exe.exe
    C:/Program Files (x86)/Intel/oneAPI/mpi/latest/bin/release/hello_exe
    C:/Program Files (x86)/Intel/oneAPI/mpi/latest/libfabric/bin/hello_exe.com
    C:/Program Files (x86)/Intel/oneAPI/mpi/latest/libfabric/bin/hello_exe.exe
    C:/Program Files (x86)/Intel/oneAPI/mpi/latest/libfabric/bin/hello_exe
    C:/Program Files (x86)/Intel/oneAPI/mpi/latest/libfabric/bin/utils/hello_exe.com
    C:/Program Files (x86)/Intel/oneAPI/mpi/latest/libfabric/bin/utils/hello_exe.exe
    C:/Program Files (x86)/Intel/oneAPI/mpi/latest/libfabric/bin/utils/hello_exe
    C:/Program Files (x86)/Intel/oneAPI/tbb/latest/redist/intel64/vc_mt/hello_exe.com
    C:/Program Files (x86)/Intel/oneAPI/tbb/latest/redist/intel64/vc_mt/hello_exe.exe
    C:/Program Files (x86)/Intel/oneAPI/tbb/latest/redist/intel64/vc_mt/hello_exe
    C:/Program Files (x86)/Intel/oneAPI/tbb/latest/redist/ia32/vc_mt/hello_exe.com
    C:/Program Files (x86)/Intel/oneAPI/tbb/latest/redist/ia32/vc_mt/hello_exe.exe
    C:/Program Files (x86)/Intel/oneAPI/tbb/latest/redist/ia32/vc_mt/hello_exe
    C:/Program Files (x86)/Intel/oneAPI/compiler/latest/windows/redist/intel64_win/compiler/hello_exe.com
    C:/Program Files (x86)/Intel/oneAPI/compiler/latest/windows/redist/intel64_win/compiler/hello_exe.exe
    C:/Program Files (x86)/Intel/oneAPI/compiler/latest/windows/redist/intel64_win/compiler/hello_exe
    C:/Program Files (x86)/Intel/oneAPI/compiler/latest/windows/redist/ia32_win/compiler/hello_exe.com
    C:/Program Files (x86)/Intel/oneAPI/compiler/latest/windows/redist/ia32_win/compiler/hello_exe.exe
    C:/Program Files (x86)/Intel/oneAPI/compiler/latest/windows/redist/ia32_win/compiler/hello_exe
    C:/Program Files/Python39/Scripts/hello_exe.com
    C:/Program Files/Python39/Scripts/hello_exe.exe
    C:/Program Files/Python39/Scripts/hello_exe
    C:/Program Files/Python39/hello_exe.com
    C:/Program Files/Python39/hello_exe.exe
    C:/Program Files/Python39/hello_exe
    C:/Windows/System32/hello_exe.com
    C:/Windows/System32/hello_exe.exe
    C:/Windows/System32/hello_exe
    C:/Windows/hello_exe.com
    C:/Windows/hello_exe.exe
    C:/Windows/hello_exe
    C:/Windows/System32/wbem/hello_exe.com
    C:/Windows/System32/wbem/hello_exe.exe
    C:/Windows/System32/wbem/hello_exe
    C:/Windows/System32/WindowsPowerShell/v1.0/hello_exe.com
    C:/Windows/System32/WindowsPowerShell/v1.0/hello_exe.exe
    C:/Windows/System32/WindowsPowerShell/v1.0/hello_exe
    C:/Windows/System32/OpenSSH/hello_exe.com
    C:/Windows/System32/OpenSSH/hello_exe.exe
    C:/Windows/System32/OpenSSH/hello_exe
    C:/Program Files (x86)/NVIDIA Corporation/PhysX/Common/hello_exe.com
    C:/Program Files (x86)/NVIDIA Corporation/PhysX/Common/hello_exe.exe
    C:/Program Files (x86)/NVIDIA Corporation/PhysX/Common/hello_exe
    C:/Program Files/Microsoft SQL Server/130/Tools/Binn/hello_exe.com
    C:/Program Files/Microsoft SQL Server/130/Tools/Binn/hello_exe.exe
    C:/Program Files/Microsoft SQL Server/130/Tools/Binn/hello_exe
    C:/Program Files/Git/cmd/hello_exe.com
    C:/Program Files/Git/cmd/hello_exe.exe
    C:/Program Files/Git/cmd/hello_exe
    C:/Program Files/PuTTY/hello_exe.com
    C:/Program Files/PuTTY/hello_exe.exe
    C:/Program Files/PuTTY/hello_exe
    C:/Program Files/Perforce/hello_exe.com
    C:/Program Files/Perforce/hello_exe.exe
    C:/Program Files/Perforce/hello_exe
    C:/Program Files/NASM/hello_exe.com
    C:/Program Files/NASM/hello_exe.exe
    C:/Program Files/NASM/hello_exe
    D:/conan_find_program/test_package/build/hello_exe.com
    D:/conan_find_program/test_package/build/hello_exe.exe
    D:/conan_find_program/test_package/build/hello_exe
    C:/Program Files/Go/bin/hello_exe.com
    C:/Program Files/Go/bin/hello_exe.exe
    C:/Program Files/Go/bin/hello_exe
    C:/Program Files/Docker/Docker/resources/bin/hello_exe.com
    C:/Program Files/Docker/Docker/resources/bin/hello_exe.exe
    C:/Program Files/Docker/Docker/resources/bin/hello_exe
    C:/ProgramData/DockerDesktop/version-bin/hello_exe.com
    C:/ProgramData/DockerDesktop/version-bin/hello_exe.exe
    C:/ProgramData/DockerDesktop/version-bin/hello_exe
    C:/Program Files/dotnet/hello_exe.com
    C:/Program Files/dotnet/hello_exe.exe
    C:/Program Files/dotnet/hello_exe
    C:/Users/my_username/.cargo/bin/hello_exe.com
    C:/Users/my_username/.cargo/bin/hello_exe.exe
    C:/Users/my_username/.cargo/bin/hello_exe
    C:/Users/my_username/AppData/Local/Microsoft/WindowsApps/hello_exe.com
    C:/Users/my_username/AppData/Local/Microsoft/WindowsApps/hello_exe.exe
    C:/Users/my_username/AppData/Local/Microsoft/WindowsApps/hello_exe
    C:/Users/my_username/AppData/Local/Programs/Microsoft VS Code/bin/hello_exe.com
    C:/Users/my_username/AppData/Local/Programs/Microsoft VS Code/bin/hello_exe.exe
    C:/Users/my_username/AppData/Local/Programs/Microsoft VS Code/bin/hello_exe
    C:/Users/my_username/AppData/Local/Google/Cloud SDK/google-cloud-sdk/bin/hello_exe.com
    C:/Users/my_username/AppData/Local/Google/Cloud SDK/google-cloud-sdk/bin/hello_exe.exe
    C:/Users/my_username/AppData/Local/Google/Cloud SDK/google-cloud-sdk/bin/hello_exe
    C:/Program Files (x86)/Microsoft Visual Studio/2019/Professional/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/bin/hello_exe.com
    C:/Program Files (x86)/Microsoft Visual Studio/2019/Professional/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/bin/hello_exe.exe
    C:/Program Files (x86)/Microsoft Visual Studio/2019/Professional/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/bin/hello_exe
    D:/vcpkg/hello_exe.com
    D:/vcpkg/hello_exe.exe
    D:/vcpkg/hello_exe
    C:/Users/my_username/go/bin/hello_exe.com
    C:/Users/my_username/go/bin/hello_exe.exe
    C:/Users/my_username/go/bin/hello_exe
    C:/Users/my_username/.dotnet/tools/hello_exe.com
    C:/Users/my_username/.dotnet/tools/hello_exe.exe
    C:/Users/my_username/.dotnet/tools/hello_exe
    D:/terraform_1.3.4_windows_amd64/hello_exe.com
    D:/terraform_1.3.4_windows_amd64/hello_exe.exe
    D:/terraform_1.3.4_windows_amd64/hello_exe
    D:/vault_1.12.1_windows_amd64/hello_exe.com
    D:/vault_1.12.1_windows_amd64/hello_exe.exe
    D:/vault_1.12.1_windows_amd64/hello_exe
    C:/Program Files/bin/hello_exe.com
    C:/Program Files/bin/hello_exe.exe
    C:/Program Files/bin/hello_exe
    C:/Program Files/sbin/hello_exe.com
    C:/Program Files/sbin/hello_exe.exe
    C:/Program Files/sbin/hello_exe
    C:/Program Files/hello_exe.com
    C:/Program Files/hello_exe.exe
    C:/Program Files/hello_exe
    C:/Program Files (x86)/bin/hello_exe.com
    C:/Program Files (x86)/bin/hello_exe.exe
    C:/Program Files (x86)/bin/hello_exe
    C:/Program Files (x86)/sbin/hello_exe.com
    C:/Program Files (x86)/sbin/hello_exe.exe
    C:/Program Files (x86)/sbin/hello_exe
    C:/Program Files (x86)/hello_exe.com
    C:/Program Files (x86)/hello_exe.exe
    C:/Program Files (x86)/hello_exe
    C:/Program Files (x86)/Microsoft Visual Studio/2019/Professional/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/sbin/hello_exe.com
    C:/Program Files (x86)/Microsoft Visual Studio/2019/Professional/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/sbin/hello_exe.exe
    C:/Program Files (x86)/Microsoft Visual Studio/2019/Professional/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/sbin/hello_exe
    C:/Program Files (x86)/Microsoft Visual Studio/2019/Professional/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/hello_exe.com
    C:/Program Files (x86)/Microsoft Visual Studio/2019/Professional/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/hello_exe.exe
    C:/Program Files (x86)/Microsoft Visual Studio/2019/Professional/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/hello_exe
    C:/Program Files/PackageTest/bin/hello_exe.com
    C:/Program Files/PackageTest/bin/hello_exe.exe
    C:/Program Files/PackageTest/bin/hello_exe
    C:/Program Files/PackageTest/sbin/hello_exe.com
    C:/Program Files/PackageTest/sbin/hello_exe.exe
    C:/Program Files/PackageTest/sbin/hello_exe
    C:/Program Files/PackageTest/hello_exe.com
    C:/Program Files/PackageTest/hello_exe.exe
    C:/Program Files/PackageTest/hello_exe

  The item was not found.



CMake Error at CMakeLists.txt:7 (find_program):
  Could not find TEST using the following names: hello_exe


-- Configuring incomplete, errors occurred!
See also "D:/conan_find_program/test_package/build/CMakeFiles/CMakeOutput.log".
ERROR: hello/0.1 (test package): Error in build() method, line 21
        cmake.configure()
        ConanException: Error 1 while executing cmake -G "Visual Studio 16 2019" -DCMAKE_TOOLCHAIN_FILE="D:/conan_find_program/test_package/build/generators/conan_toolchain.cmake" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" "D:\conan_find_program\test_package\."
```

**WORKING (1.53)**
```
D:\conan_find_program>conan create .
Exporting package recipe
hello/0.1 exports_sources: Copied 1 '.txt' file: CMakeLists.txt
hello/0.1 exports_sources: Copied 2 '.cpp' files: hello.cpp, main.cpp
hello/0.1 exports_sources: Copied 1 '.h' file: hello.h
hello/0.1: The stored package has not changed
hello/0.1: Using the exported files summary hash as the recipe revision: 0dbe589a6e52ddaa9ab052a116cde54b
hello/0.1: Exported revision: 0dbe589a6e52ddaa9ab052a116cde54b
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=Visual Studio
compiler.runtime=MD
compiler.version=16
os=Windows
os_build=Windows
[options]
[build_requires]
[env]

hello/0.1: Forced build from source
hello/0.1 (test package): Installing package
Requirements
    hello/0.1 from local cache - Cache
Packages
    hello/0.1:3fb49604f9c2f729b85ba3115852006824e72cab - Build

Installing (downloading, building) binaries...
hello/0.1: Copying sources to build folder
hello/0.1: Building your package in C:\Users\my_username\.conan\data\hello\0.1\_\_\build\3fb49604f9c2f729b85ba3115852006824e72cab
hello/0.1: Generator txt created conanbuildinfo.txt
hello/0.1: Calling generate()
hello/0.1: WARN: Using the new toolchains and generators without specifying a build profile (e.g: -pr:b=default) is discouraged and might cause failures and unexpected behavior
hello/0.1: Preset 'default' added to CMakePresets.json. Invoke it manually using 'cmake --preset default'
hello/0.1: If your CMake version is not compatible with CMakePresets (<3.19) call cmake like: 'cmake <path> -G "Visual Studio 16 2019" -DCMAKE_TOOLCHAIN_FILE=C:\Users\my_username\.conan\data\hello\0.1\_\_\build\3fb49604f9c2f729b85ba3115852006824e72cab\build\generators\conan_toolchain.cmake -DCMAKE_POLICY_DEFAULT_CMP0091=NEW'
hello/0.1: Aggregating env generators
hello/0.1: Calling build()
hello/0.1: CMake command: cmake -G "Visual Studio 16 2019" -DCMAKE_TOOLCHAIN_FILE="C:/Users/my_username/.conan/data/hello/0.1/_/_/build/3fb49604f9c2f729b85ba3115852006824e72cab/build/generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="C:/Users/my_username/.conan/data/hello/0.1/_/_/package/3fb49604f9c2f729b85ba3115852006824e72cab" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" "C:\Users\my_username\.conan\data\hello\0.1\_\_\build\3fb49604f9c2f729b85ba3115852006824e72cab\."
-- Using Conan toolchain: C:/Users/my_username/.conan/data/hello/0.1/_/_/build/3fb49604f9c2f729b85ba3115852006824e72cab/build/generators/conan_toolchain.cmake
-- Selecting Windows SDK version 10.0.19041.0 to target Windows 10.0.19044.
-- The CXX compiler identification is MSVC 19.29.30146.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Professional/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_POLICY_DEFAULT_CMP0091


-- Build files have been written to: C:/Users/my_username/.conan/data/hello/0.1/_/_/build/3fb49604f9c2f729b85ba3115852006824e72cab/build
hello/0.1: CMake command: cmake --build "C:\Users\my_username\.conan\data\hello\0.1\_\_\build\3fb49604f9c2f729b85ba3115852006824e72cab\build" --config Release
Microsoft (R) Build Engine version 16.11.2+f32259642 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

  Checking Build System
  Building Custom Rule C:/Users/my_username/.conan/data/hello/0.1/_/_/build/3fb49604f9c2f729b85ba3115852006824e72cab/CMakeLists.txt
  hello.cpp
  main.cpp
  hello_exe.vcxproj -> C:\Users\my_username\.conan\data\hello\0.1\_\_\build\3fb49604f9c2f729b85ba3115852006824e72cab\build\Release\hello_exe.
  exe
  Building Custom Rule C:/Users/my_username/.conan/data/hello/0.1/_/_/build/3fb49604f9c2f729b85ba3115852006824e72cab/CMakeLists.txt
  hello.cpp
  hello_lib.vcxproj -> C:\Users\my_username\.conan\data\hello\0.1\_\_\build\3fb49604f9c2f729b85ba3115852006824e72cab\build\Release\hello_lib.
  lib
  Building Custom Rule C:/Users/my_username/.conan/data/hello/0.1/_/_/build/3fb49604f9c2f729b85ba3115852006824e72cab/CMakeLists.txt
hello/0.1: Package '3fb49604f9c2f729b85ba3115852006824e72cab' built
hello/0.1: Build folder C:\Users\my_username\.conan\data\hello\0.1\_\_\build\3fb49604f9c2f729b85ba3115852006824e72cab\build
hello/0.1: Generated conaninfo.txt
hello/0.1: Generated conanbuildinfo.txt
hello/0.1: Generating the package
hello/0.1: Package folder C:\Users\my_username\.conan\data\hello\0.1\_\_\package\3fb49604f9c2f729b85ba3115852006824e72cab
hello/0.1: Calling package()
hello/0.1: CMake command: cmake --install "C:\Users\my_username\.conan\data\hello\0.1\_\_\build\3fb49604f9c2f729b85ba3115852006824e72cab\build" --config Release --prefix "C:/Users/my_username/.conan/data/hello/0.1/_/_/package/3fb49604f9c2f729b85ba3115852006824e72cab"
-- Installing: C:/Users/my_username/.conan/data/hello/0.1/_/_/package/3fb49604f9c2f729b85ba3115852006824e72cab/bin/hello_exe.exe
-- Installing: C:/Users/my_username/.conan/data/hello/0.1/_/_/package/3fb49604f9c2f729b85ba3115852006824e72cab/lib/hello_lib.lib
-- Installing: C:/Users/my_username/.conan/data/hello/0.1/_/_/package/3fb49604f9c2f729b85ba3115852006824e72cab/include/hello.h
hello/0.1 package(): Packaged 1 '.exe' file: hello_exe.exe
hello/0.1 package(): Packaged 1 '.h' file: hello.h
hello/0.1 package(): Packaged 1 '.lib' file: hello_lib.lib
hello/0.1: Package '3fb49604f9c2f729b85ba3115852006824e72cab' created
hello/0.1: Created package revision 1e47a8068dafe50aa73e3c864b840650
hello/0.1 (test package): Generator txt created conanbuildinfo.txt
hello/0.1 (test package): WARN: Using the new toolchains and generators without specifying a build profile (e.g: -pr:b=default) is discouraged and might cause failures and unexpected behavior
hello/0.1 (test package): Generator 'CMakeDeps' calling 'generate()'
hello/0.1 (test package): WARN: Using the new toolchains and generators without specifying a build profile (e.g: -pr:b=default) is discouraged and might cause failures and unexpected behavior
hello/0.1 (test package): Generator 'CMakeToolchain' calling 'generate()'
hello/0.1 (test package): Preset 'default' added to CMakePresets.json. Invoke it manually using 'cmake --preset default'
hello/0.1 (test package): If your CMake version is not compatible with CMakePresets (<3.19) call cmake like: 'cmake <path> -G "Visual Studio 16 2019" -DCMAKE_TOOLCHAIN_FILE=D:\conan_find_program\test_package\build\generators\conan_toolchain.cmake -DCMAKE_POLICY_DEFAULT_CMP0091=NEW'
hello/0.1 (test package): Aggregating env generators
hello/0.1 (test package): Generated conaninfo.txt
hello/0.1 (test package): Generated graphinfo
Using lockfile: 'D:\conan_find_program\test_package\build\generators/conan.lock'
Using cached profile from lockfile
hello/0.1 (test package): Calling build()
hello/0.1 (test package): CMake command: cmake -G "Visual Studio 16 2019" -DCMAKE_TOOLCHAIN_FILE="D:/conan_find_program/test_package/build/generators/conan_toolchain.cmake" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" "D:\conan_find_program\test_package\."
-- Using Conan toolchain: D:/conan_find_program/test_package/build/generators/conan_toolchain.cmake
-- Selecting Windows SDK version 10.0.19041.0 to target Windows 10.0.19044.
-- The CXX compiler identification is MSVC 19.29.30146.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Professional/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Conan: Target declared 'hello::hello'
CMake Debug Log at CMakeLists.txt:7 (find_program):
  find_program called with the following settings:

    VAR: TEST
    NAMES: "hello_exe"
    Documentation: Path to a program.
    Framework
      Only Search Frameworks: 0
      Search Frameworks Last: 0
      Search Frameworks First: 0
    AppBundle
      Only Search AppBundle: 0
      Search AppBundle Last: 0
      Search AppBundle First: 0
    CMAKE_FIND_USE_CMAKE_PATH: 1
    CMAKE_FIND_USE_CMAKE_ENVIRONMENT_PATH: 1
    CMAKE_FIND_USE_SYSTEM_ENVIRONMENT_PATH: 1
    CMAKE_FIND_USE_CMAKE_SYSTEM_PATH: 1

  find_program considered the following locations:

    C:/Users/my_username/.conan/data/hello/0.1/_/_/package/3fb49604f9c2f729b85ba3115852006824e72cab/bin/hello_exe.com

  The item was found at

    C:/Users/my_username/.conan/data/hello/0.1/_/_/package/3fb49604f9c2f729b85ba3115852006824e72cab/bin/hello_exe.exe



-- hello_exe: C:/Users/my_username/.conan/data/hello/0.1/_/_/package/3fb49604f9c2f729b85ba3115852006824e72cab/bin/hello_exe.exe
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_POLICY_DEFAULT_CMP0091


-- Build files have been written to: D:/conan_find_program/test_package/build
hello/0.1 (test package): CMake command: cmake --build "D:\conan_find_program\test_package\build" --config Release
Microsoft (R) Build Engine version 16.11.2+f32259642 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

  Checking Build System
  Building Custom Rule D:/conan_find_program/test_package/CMakeLists.txt
  example.cpp
  example.vcxproj -> D:\conan_find_program\test_package\build\Release\example.exe
  Building Custom Rule D:/conan_find_program/test_package/CMakeLists.txt
hello/0.1 (test package): Running test()
hello/0.1: Hello World Release!
  hello/0.1: _M_X64 defined
  hello/0.1: _MSC_VER1929
  hello/0.1: _MSVC_LANG201402
  hello/0.1: __cplusplus199711
```

--- a/conan/tools/cmake/cmakedeps/templates/target_configuration.py
+++ b/conan/tools/cmake/cmakedeps/templates/target_configuration.py
@@ -78,11 +78,6 @@ class TargetConfigurationTemplate(CMakeDepsFileTemplate):
 
         # FIXME: What is the result of this for multi-config? All configs adding themselves to path?
         set(CMAKE_MODULE_PATH {{ '${' }}{{ pkg_name }}_BUILD_DIRS{{ config_suffix }}} {{ '${' }}CMAKE_MODULE_PATH})
-        if(NOT CMAKE_CROSSCOMPILING)
-            # Kept for compatibility in Conan 1.x when using a single profile, this should
-            # not be present in Conan 2.0
-            set(CMAKE_PREFIX_PATH {{ '${' }}{{ pkg_name }}_BUILD_DIRS{{ config_suffix }}} {{ '${' }}CMAKE_PREFIX_PATH})
-        endif()
         {% if not components_names %}
 
         ########## GLOBAL TARGET PROPERTIES {{ configuration }} ########################################

--- a/conan/tools/cmake/cmakedeps/templates/target_configuration.py
+++ b/conan/tools/cmake/cmakedeps/templates/target_configuration.py
@@ -78,7 +78,11 @@ class TargetConfigurationTemplate(CMakeDepsFileTemplate):
 
         # FIXME: What is the result of this for multi-config? All configs adding themselves to path?
         set(CMAKE_MODULE_PATH {{ '${' }}{{ pkg_name }}_BUILD_DIRS{{ config_suffix }}} {{ '${' }}CMAKE_MODULE_PATH})
-
+        if(NOT CMAKE_CROSSCOMPILING)
+            # Kept for compatibility in Conan 1.x when using a single profile, this should
+            # not be present in Conan 2.0
+            set(CMAKE_PREFIX_PATH {{ '${' }}{{ pkg_name }}_BUILD_DIRS{{ config_suffix }}} {{ '${' }}CMAKE_PREFIX_PATH})
+        endif()
         {% if not components_names %}
 
         ########## GLOBAL TARGET PROPERTIES {{ configuration }} ########################################

--- a/conan/tools/cmake/toolchain/blocks.py
+++ b/conan/tools/cmake/toolchain/blocks.py
@@ -532,6 +532,7 @@ class FindFiles(Block):
         host_build_paths_root = []
         host_build_paths_noroot = []
         host_lib_paths = []
+        host_bin_paths = [] # Only to be used when only one profile is present
         host_framework_paths = []
         host_include_paths = []
         for req in host_req:
@@ -547,6 +548,7 @@ class FindFiles(Block):
                 host_build_paths_root = []
                 host_build_paths_noroot.extend(p for p in cppinfo.builddirs)
             host_lib_paths.extend(cppinfo.libdirs)
+            host_bin_paths.extend(cppinfo.bindirs)
             if is_apple_:
                 host_framework_paths.extend(cppinfo.frameworkdirs)
             host_include_paths.extend(cppinfo.includedirs)
@@ -559,6 +561,8 @@ class FindFiles(Block):
             cppinfo = req.cpp_info.aggregated_components()
             build_build_paths.extend(cppinfo.builddirs)
             build_bin_paths.extend(cppinfo.bindirs)
+        has_build_context = hasattr(self._conanfile, "settings_build") # always true in Conan 2.0
+        bin_paths = build_bin_paths if has_build_context else host_bin_paths
 
         return {
             "find_package_prefer_config": find_package_prefer_config,
@@ -566,7 +570,7 @@ class FindFiles(Block):
             "host_build_paths_root": self._join_paths(host_build_paths_root),
             "host_build_paths_noroot": self._join_paths(host_build_paths_noroot),
             "build_build_paths": self._join_paths(build_build_paths),
-            "cmake_program_path": self._join_paths(build_bin_paths),
+            "cmake_program_path": self._join_paths(bin_paths),
             "cmake_library_path": self._join_paths(host_lib_paths),
             "cmake_framework_path": self._join_paths(host_framework_paths),
             "cmake_include_path": self._join_paths(host_include_paths),

--- a/conans/test/functional/toolchains/cmake/test_cmake_toolchain.py
+++ b/conans/test/functional/toolchains/cmake/test_cmake_toolchain.py
@@ -1188,7 +1188,8 @@ def test_cmake_toolchain_vars_when_option_declared():
 
 
 @pytest.mark.tool_cmake
-def test_find_program_for_tool_requires():
+@pytest.mark.parametrize("single_profile", [True, False])
+def test_find_program_for_tool_requires(single_profile):
     """Test that the same reference can be both a tool_requires and a regular requires,
     and that find_program (executables) and find_package (libraries) find the correct ones
     when cross building.
@@ -1240,8 +1241,11 @@ def test_find_program_for_tool_requires():
                 "build_profile": build_profile
                 })
 
-    client.run("create . -pr:b build_profile -pr:h build_profile")
-    client.run("create . -pr:b build_profile -pr:h host_profile")
+    if single_profile:
+        client.run("create . --profile host_profile")
+    else:
+        client.run("create . -pr:b build_profile -pr:h build_profile")
+        client.run("create . -pr:b build_profile -pr:h host_profile")
 
     conanfile_consumer = textwrap.dedent("""
         from conan import ConanFile
@@ -1256,7 +1260,7 @@ def test_find_program_for_tool_requires():
                 self.requires("foobar/1.0")
 
             def build_requirements(self):
-                self.tool_requires("foobar/1.0")
+               self.tool_requires("foobar/1.0")
     """)
 
     cmakelists_consumer = textwrap.dedent("""
@@ -1266,6 +1270,9 @@ def test_find_program_for_tool_requires():
         find_program(FOOBIN_EXECUTABLE foobin)
         message("foobin executable: ${FOOBIN_EXECUTABLE}")
         message("foobar include dir: ${foobar_INCLUDE_DIR}")
+        if(NOT FOOBIN_EXECUTABLE)
+          message(FATAL_ERROR "FOOBIN executable not found")
+        endif()
     """)
 
     client.save({
@@ -1273,17 +1280,24 @@ def test_find_program_for_tool_requires():
         "CMakeLists.txt": cmakelists_consumer,
         "host_profile": host_profile,
         "build_profile": build_profile}, clean_first=True)
-    client.run("install conanfile_consumer.py pkg/0.1@ -g CMakeToolchain -g CMakeDeps -pr:b build_profile -pr:h host_profile")
-
-    build_context_package_id = "581814504b2e960b35df487e5bdb32b1ecf02253"
-    host_context_package_id = "bf544cd3bc20b82121fd76b82eacbb36d75fa167"
+    
+    if single_profile:
+        profiles = "-pr host_profile"
+    else:
+        profiles = "-pr:b build_profile -pr:h host_profile"
+    
+    client.run(f"install conanfile_consumer.py pkg/0.1@ -g CMakeToolchain -g CMakeDeps {profiles}")
 
     with client.chdir("build"):
         client.run_command("cmake .. -DCMAKE_TOOLCHAIN_FILE=Release/generators/conan_toolchain.cmake -DCMAKE_BUILD_TYPE=Release")
         # Verify binary executable is found from build context package,
         # and library comes from host context package
-        assert f"package/{build_context_package_id}/bin/foobin" in client.out
-        assert f"package/{host_context_package_id}/include" in client.out
+
+        if not single_profile:
+            build_context_package_id = "581814504b2e960b35df487e5bdb32b1ecf02253"
+            host_context_package_id = "bf544cd3bc20b82121fd76b82eacbb36d75fa167"
+            assert f"package/{build_context_package_id}/bin/foobin" in client.out
+            assert f"package/{host_context_package_id}/include" in client.out
 
 
 @pytest.mark.tool_pkg_config

